src/mavsdk_server/src/mavsdk_server.cpp

```diff
@@ -15,13 +15,19 @@ class MavsdkServer::Impl {
 
     bool connect(const std::string& connection_url)
     {
-        _connection_initiator.start(_mavsdk, connection_url);
+        if (_mavsdk == nullptr) {
+            _mavsdk = std::make_unique<mavsdk::Mavsdk>();
+        }
+        _connection_initiator.start(*_mavsdk, connection_url);
         return _connection_initiator.wait();
     }
 
     int startGrpcServer(const int port)
     {
-        _server = std::make_unique<GrpcServer>(_mavsdk);
+        if (_mavsdk == nullptr) {
+            _mavsdk = std::make_unique<mavsdk::Mavsdk>();
+        }
+        _server = std::make_unique<GrpcServer>(*_mavsdk);
```

Comment:
Does that work? What happens if `_mavsdk` has not discovered any System when `_server` is created? It's supposed to take the first system and talk to it, if I remember correctly. The reason why `connect()` is blocking is that `_mavsdk_` needs to have a `System` discovered before the gRPC server can work (otherwise what should the gRPC server do with requests?)

Reply:
This is like before, isn't it? You create Mavsdk but you lazily create the plugins once you require them and have a sytem to pass to them

Reply:
Well it has changed back and forth a few times over the years, but it seems like the lazy plugins now allow that again. My concern is that every time it changed, it broke the API (`mavsdk_server.run()` was blocking, then not, then blocking again, etc).

Now with the solution throwing `NoSystem`, making it non-blocking means that the user now has to try repeatedly and start working once a system is there. So I honestly like the blocking call better, I think :thinking:.

Reply:
Did I change that though? Maybe I don't understand what I changed.

Reply:
Ooooh, I get it now. You essentially just changed it such that `MAVSDK` is a pointer that you can reset in order to "stop" it. Well I guess that works, too. I would like it better if `stop()` called `mavsdk.stop()` or `mavsdk.remove_all_connections()`, but probably this works :+1:

Reply:
Yes, I will implement, so that you can remove connections.